runtime/src/iree/hal/drivers/cuda/nccl_channel.c

```diff
@@ -67,24 +78,21 @@ iree_status_t iree_hal_cuda_nccl_channel_create(
   IREE_TRACE_ZONE_APPEND_VALUE(z0, rank);
   IREE_TRACE_ZONE_APPEND_VALUE(z0, count);
 
-  // TODO(#9580): actually use nccl to create a communicator.
-  // Something like:
-  //  ncclConfig_t config = NCCL_CONFIG_INITIALIZER;
-  //  config.blocking = 0;
-  //  syms->ncclCommInitRankConfig(&comm, count, *id, rank, &config);
-  // NOTE: CHECK ERRORS! we can safely return here as we haven't allocated the
-  // channel wrapper yet.
   ncclComm_t comm = NULL;
-  if (!comm) {
+  ncclConfig_t config = NCCL_CONFIG_INITIALIZER;
+  config.blocking = 1;  // FIXME: use async to check a timeout
```

Comment:
Is this overlooked or it is meant to be fixed in a later PR?

Reply:
It is meant to be fixed later.

Reply:
Let me take a look at it now since I did the async channel destroy below.

Reply:
Well, the nonblocking channel creation didn't work as expected, so I left a to-do in the channel destroy function. Nonblocking should be revisited later.